File: easy/136_racing_chars.py
import sys

prev_pos = None
for line in (l.strip() for l in open(sys.argv[1])):
    pos = line.find('C')
    if pos == -1:
        pos = line.find('_')
    if prev_pos == pos or prev_pos is None:
        char = '|'
    elif prev_pos < pos:
        char = '\\'
    else:
        char = '/'
    print(line[:pos] + char + line[pos+1:])
    prev_pos = pos

File: easy/136_racing_chars.c
#include <stdio.h>
#include <strings.h>

int main(int argc, const char * argv[]) {
    FILE *file = fopen(argv[1], "r");
    char line[1024];
    char *pos;
    char *prev_pos = (char *)NULL;
    char c;

    while (fgets(line, 1024, file)) {
        if ((pos = index(line, 'C')) == NULL)
            pos = index(line, '_');

        if ((prev_pos == pos) || (prev_pos == NULL))
            *pos = '|';
        else if (prev_pos < pos)
            *pos = '\\';
        else
            *pos = '/';

        prev_pos = pos;
        printf(line);
    }

    return 0;
}
